main.c

```diff
@@ -0,0 +1,261 @@
+#include "main.h"
+
+struct Node* CreateNode ()
+{
+  struct Node* newNode = (struct Node*)malloc(sizeof(struct Node));
+  if (!newNode) {
+    printf("Memory didn't allocated, cannot continue\n");
+    exit(1);
```

Comment:
This way you don't give the user of your container any chance to free any other resources he might have.